### PR TITLE
i06 unreachable return statement

### LINKED ISSUE #2077: I06: Unreachable return statement
Affected Code: LightClient.sol (line 424)

Summary: The current implementation of getHotShotCommitment includes
an unreachable return statement.

Suggestion: We recommend to refactor the unreachable return state-
ment.

### PR DESCRIPTION
**Pull request by alysiahuggins**
Closes #2077 

### This PR:
- removes unreachable return statement
- modify the function to return the blockcomm root when the provided `hotShotBlockHeight` is equal to the blockHeight in the `hotShotCommitments` array
- modified tests to test the above modification

### Key places to review:
- the function
- the function comments

### How to test this PR:
- `forge test --match-test test_GetHotShotCommitmentValid`
- `forge test --match-test test_revertWhenGetHotShotCommitmentInvalidHeight`

### CODE REVIEW

File: contracts/test/LightClient.t.sol [hunk 1]
```diff
@@ -1153,9 +1153,19 @@ contract LightClient_HotShotCommUpdatesTest is LightClientCommonTest {
             lc.getHotShotCommitment(newState.blockHeight - 1);
         assertEqBN254(blockComm, newState.blockCommRoot);
         assertEq(blockHeight, newState.blockHeight);
+
+        // Test for max hotShotBlockHeight stored on contract
+        // Get the highest HotShot blockheight recorded
+        uint256 numCommitments = lc.getStateHistoryCount();
+        (,, blockHeight,) = lc.stateHistoryCommitments(numCommitments - 1);
+        lc.getHotShotCommitment(blockHeight);
```
Comment (Sneh1999) [inline contracts/test/LightClient.t.sol:1161]: should we have an asset statement here? seems like we are only calling the function and not checking for anything

File: contracts/test/LightClient.t.sol [hunk 2]
```diff
@@ -1153,9 +1153,19 @@ contract LightClient_HotShotCommUpdatesTest is LightClientCommonTest {
...
+
+        // Get the smallest HotShot blockheight recorded
+        (,, blockHeight,) = lc.stateHistoryCommitments(0);
+        lc.getHotShotCommitment(blockHeight);
```
Comment (Sneh1999) [inline contracts/test/LightClient.t.sol:1165]: same here!

File: contracts/test/LightClient.t.sol (hunk 1 above)
Comment (alysiahuggins) [reply contracts/test/LightClient.t.sol]: https://github.com/EspressoSystems/espresso-sequencer/commit/52629295adfdc947c8648cb8c96ac2f7643d7f1b

File: contracts/test/LightClient.t.sol (hunk 2 above)
Comment (alysiahuggins) [reply contracts/test/LightClient.t.sol]: https://github.com/EspressoSystems/espresso-sequencer/commit/52629295adfdc947c8648cb8c96ac2f7643d7f1b

**Review by Sneh1999 — APPROVED** [review]
LGTM!

**Pull request merged**